File: HPWorkpath_v1.6.3/Samples/ExampleAPIServices/source/AuthorizationSample/src/com/hp/workpath/sample/authorization/receiver/ChangeNotificationReceiver.java
// Copyright 2025 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.authorization.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import com.hp.workpath.api.authorization.ChangeNotificationEventCode;
import com.hp.workpath.api.authorization.ChangeNotificationEventData;
import com.hp.workpath.sample.authorization.MainActivity;

public class ChangeNotificationReceiver extends BroadcastReceiver {

    private static final String ACTION_CHANGE_NOTIFICATION = "com.hp.workpath.action.AUTHORIZATION_CHANGE_NOTIFICATION";

    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();
        if (action != null && action.equals(ACTION_CHANGE_NOTIFICATION)) {
            ChangeNotificationEventData data = intent.getParcelableExtra("changeNotificationEventData");
            Log.i(MainActivity.TAG, "ChangeNotificationReceiver: eventCode=" + data.getEventCode() + ", timestamp=" + data.getTimestamp());

            if (data.getEventCode().equals(ChangeNotificationEventCode.PERMISSION_ADDED)) {
                Log.i(MainActivity.TAG, "Permissions have changed");
            } else if (data.getEventCode().equals(ChangeNotificationEventCode.PROXY_CONFIGURATION_CHANGED)) {
                Log.i(MainActivity.TAG, "Configuration has changed");
            }
        }
    }
}

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleAPIServices/source/AuthorizationSample/src/com/hp/workpath/sample/authorization/receiver/ChangeNotificationReceiver.kt
// Copyright 2025 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.authorization.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import com.hp.workpath.api.authorization.ChangeNotificationEventCode
import com.hp.workpath.api.authorization.ChangeNotificationEventData
import com.hp.workpath.sample.authorization.MainActivity

class ChangeNotificationReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context?, intent: Intent?) {
        val action = intent?.action
        if (action == ACTION_CHANGE_NOTIFICATION) {
            val data =
                intent.getParcelableExtra<ChangeNotificationEventData>("changeNotificationEventData")
            Log.i(
                MainActivity.TAG,
                "ChangeNotificationReceiver: eventCode=${data?.eventCode}, timestamp=${data?.timestamp}"
            )
            when (data?.eventCode) {
                ChangeNotificationEventCode.PERMISSION_ADDED.value() -> {
                    // Handle permission added event
                    Log.i(MainActivity.TAG, "Permissions have changed")
                }

                ChangeNotificationEventCode.PROXY_CONFIGURATION_CHANGED.value() -> {
                    // Handle proxy configuration changed event
                    Log.i(MainActivity.TAG, "Configuration has changed")
                }

                else -> {
                    // Handle unknown event code
                }
            }
        }
    }

    companion object {
        private const val ACTION_CHANGE_NOTIFICATION =
            "com.hp.workpath.action.AUTHORIZATION_CHANGE_NOTIFICATION"
    }
}